packages/spanner/src/database.js

```diff
@@ -713,45 +713,43 @@ Database.prototype.runStream = function(query, options) {
  * });
  *
  * //-
- * // If the callback is omitted, we'll return a Promise.
+ * // To use promises, simply return a Promise.
  * //-
- * database.runTransaction()
- *   .then(function(data) {
- *     var transaction = data[0];
- *
- *     // Run a transactional query.
- *     return transaction.run('SELECT * FROM Singers')
- *       .then(function() {
- *         // Queue a mutation (note there is no callback passed to `insert`).
- *         transaction.insert('Singers', {
- *           SingerId: 'Id3b',
- *           Name: 'Joe West'
- *         });
- *
- *         // Commit the transaction.
- *         return transaction.commit();
- *       });
- *   })
- *   .then(function() {
- *     // Transaction committed successfully.
+ * database.runTransaction(function(err, transaction) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // Run a transactional query.
+ *   return transaction.run('SELECT * FROM Singers').then(function() {
+ *     // Queue a mutation (note there is no callback passed to `insert`).
+ *     transaction.insert('Singers', {
+ *       SingerId: 'Id3b',
+ *       Name: 'Joe West'
+ *     });
+ *     // Commit the transaction.
+ *     return transaction.commit();
```

Comment:
@lukesneeringer In  your example you are not passing a callable to runTransaction. As far as I can tell the API is that the runTransaction takes in a callable that returns a Promise. User should resolve that Promise only when he is done with the txn. So your example would look like this:

```es6
transaction.runTransaction(function(error, txn) {
  // Note that insert and updates just buffer the mutations in memory and do not involve any IO so 
 // I think they dont return a Promise.
  txn.insert('...data to insert...'))
  txn.update('...data to change...'))
  txn.insert('...other data to insert...'))
  // If you wanted to also do multiple queries you could do this.
  return txn.run("Select * from Singer")
    .then(txn => txn.run("Select * from Singers")
    .then(txn => txn.commit());
})
  .catch(err => handle error /* this runs if ANYTHING above fails */);
```

Also note that user does not call rollback himself, it automatically gets rolled back by the client library if there is some non retryable error